src/components/structures/RoomView.tsx

```diff
@@ -2098,6 +2113,52 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
 
         const showChatEffects = SettingsStore.getValue('showChatEffects');
 
+        // Decide what to show in the main split
+        let mainSplitBody;
+        if (SettingsStore.getValue("feature_maximised_widgets")) {
+            switch (this.state.mainSplitContentType) {
+                case MainSplitContentType.Timeline:
+                    mainSplitBody = <React.Fragment>
+                        { auxPanel }
+                        <div className={timelineClasses}>
+                            { fileDropTarget }
+                            { topUnreadMessagesBar }
+                            { jumpToBottom }
+                            { messagePanel }
+                            { searchResultsPanel }
+                        </div>
+                        { statusBarArea }
+                        { previewBar }
+                        { messageComposer }
+                    </React.Fragment>;
+                    break;
+                case MainSplitContentType.MaximisedWidget:
+                    mainSplitBody = <AppMaximisedDrawer
+                        room={this.state.room}
+                        userId={this.context.credentials.userId}
+                        resizeNotifier={this.props.resizeNotifier}
+                        showApps={true}
+                    />;
+                    break;
+                // TODO-video MainSplitContentType.Video:
+                //     break;
+            }
+        } else {
+            mainSplitBody = <React.Fragment>
```

Comment:
I believe this is identical to the what's in the `MainSplitContentType.Timeline` case above, right? Duplication causes some risk that a future change updates only one and not the other.

While you can capture it in constant and share it, I believe the feature check is unnecessary here, since some other path that's setting the state presumably also does a feature check. Removing the feature check would allow removing the else branch, so we'd be back down to one copy.

Reply:
Well spotted, it indeed makes no sense to do the feature check if the maximised case can only be hit when the feature is activated...
But I need to be very careful when I add the loading from room state to add a check there. Otherwise this branch can be hit on an element instance where the flag is deactivated when someone shares the maximised state over the room state.

Reply:
Maybe a better way that still protects things properly is for the paths _in this component_ which set the `mainSplitContentType` to check the feature? That way the render path remains easy to read (no feature check there) and the component is well protected from going to an unexpected state.

Reply:
This is now done without duplicate code.